pkg.pl

```diff
@@ -269,10 +255,10 @@
 % Execute the logical plan
 ensure_dependencies(Plan, Success) :-
     phrase(physical_plan(Plan), Ls),
-    if_(
-        Ls = [],
-        D_String = Ls,
-        Ls = [_|D_String]
+    (
+        Ls == [],
+        D_String = Ls
+        ; Ls = [_|D_String]
```

Comment:
This changes the meaning of the clause. On backtracking, the other branch will be tried.